radicle-cli/src/commands/patch/checkout.rs

```diff
@@ -0,0 +1,43 @@
+use crate::terminal as term;
+use radicle::cob::patch::{self, PatchId};
+use radicle::git;
+use radicle::profile::Profile;
+use radicle::storage::git::Repository;
+
+pub fn run(
+    storage: &Repository,
+    profile: &Profile,
+    git_workdir: &git::raw::Repository,
+    patch_id: &PatchId,
+) -> anyhow::Result<()> {
+    let patchs = patch::Patches::open(profile.public_key, storage)?;
+    let Some(patch) = patchs.get(patch_id)? else {
+        anyhow::bail!("Patch `{patch_id}` not found");
+    };
+
+    let spinner = term::spinner("Performing patch checkout...");
+
+    // Getting the patch obj!
+    let patch_haed = patch.head();
+    let obj = git_workdir.revparse_single(&patch_haed.to_string())?;
+
+    let branch_name = format!("patch/{}", term::format::cob(patch_id));
+    // checkout the patch in a new branch!
+    let commit = if let Some(commit) = obj.as_commit() {
+        commit
+    } else {
+        anyhow::bail!("Unable to convert object to a commit.");
+    };
```

Comment:
```suggestion
    let commit = obj.as_commit().ok_or_else(|| anyhow::bail!("BUG: patch did not point to a commit"));
```

Reply:
same of there https://github.com/radicle-dev/heartwood/pull/304#discussion_r1106192804